app/code/Magento/QuoteGraphQl/Model/Resolver/AddSimpleProductsToCart.php

```diff
@@ -58,12 +67,20 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
             throw new GraphQlInputException(__('Required parameter "cart_items" is missing'));
         }
         $cartItems = $args['input']['cart_items'];
-
         $storeId = (int)$context->getExtensionAttributes()->getStore()->getId();
+
+        $lockName = 'cart_processing_lock_' . $maskedCartId;
+        while ($this->lockManager->isLocked($lockName)) {
```

Comment:
Please add there some stopping mechanism for case, when it will not be unlocked during let's say 10 seconds.

Reply:
Oh... I see that these changes were already delivered to 2.4-develop as is.
I don't think it's a good enough solution as it might cause an infinite loop. Could you still introduce the requested logic and merge it aswell?